src/PoolManager.sol

```diff
@@ -185,16 +185,20 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         IPoolManager.ModifyPositionParams memory params,
         bytes calldata hookData
     ) external override noDelegateCall onlyByLocker returns (BalanceDelta delta) {
+        PoolId id = key.toId();
+        if (pools[id].isNotInitialized()) revert PoolNotInitialized();
+
         if (key.hooks.shouldCallBeforeModifyPosition()) {
-            if (
-                key.hooks.beforeModifyPosition(msg.sender, key, params, hookData)
-                    != IHooks.beforeModifyPosition.selector
-            ) {
-                revert Hooks.InvalidHookResponse();
+            bytes4 selector = key.hooks.beforeModifyPosition(msg.sender, key, params, hookData);
+            if (selector != IHooks.beforeModifyPosition.selector) {
+                if (key.hooks.isValidNoOpCall(selector)) {
+                    return BalanceDelta.wrap(0);
```

Comment:
Maybe some return indication here to tell the caller that custom accounting was used?
Either a special sentinel value (though that could probs always conflict with a truly possible balancedelta) or a separate boolean?

Reply:
Hmm interesting.. I like that idea. That would definitely make writing tests (or any integrating contract) a little easier so I'm in favor.

Reply:
Changed it to return the maximum int256... it feels unlikely that a legit trade would ever return that?